integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/QueryStringPredicateSpecificsIT.java

```diff
@@ -226,6 +233,204 @@ void enablePositionIncrements_default() {
+						.fetchAllHits()
+		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1" );
+
+		assertThatThrownBy( () -> scope.query()
+				.where( f -> f.queryString().field( field )
+						.matching( String.format( Locale.ROOT, "\"%s %s\"", value1, value2 ) ) )
+				.fetchAllHits()
+		).isInstanceOf( SearchException.class );
+
+		assertThatThrownBy( () -> scope.query()
+				.where( f -> f.queryString().field( field )
+						.matching( String.format( Locale.ROOT, "\"%s %s\"~10", value1, value2 ) ) )
+				.fetchAllHits()
+		).isInstanceOf( SearchException.class );
+
+		assertThatThrownBy( () -> scope.query()
+				.where( f -> f.queryString().field( field ).matching( String.format( Locale.ROOT, "%s~10", value1 ) ) )
+				.fetchAllHits()
+		).isInstanceOf( SearchException.class );
+
+		//  "reason": "Can only use prefix queries on keyword, text and wildcard fields - not on [integer] which is of type [long]",
+		assertThatThrownBy( () -> scope.query()
+				.where( f -> f.queryString().field( field ).matching( String.format( Locale.ROOT, "%s*", value1 ) ) )
+				.fetchAllHits()
+		).isInstanceOf( SearchException.class );
+
+		if ( TckConfiguration.get().getBackendFeatures().queryStringFailOnPatternQueries() ) {
+			assertThatThrownBy( () -> scope.query()
+					.where( f -> f.queryString().field( field ).matching( String.format( Locale.ROOT, "/%s/", value2 ) ) )
+					.fetchAllHits()
+			).isInstanceOf( SearchException.class );
+		}
+
+		assertThatThrownBy( () -> scope.query()
+				.where( f -> f.queryString().field( field ).matching( String.format( Locale.ROOT, "%s?", value2 ) ) )
+				.fetchAllHits()
+		).isInstanceOf( SearchException.class );
+
+		assertThatThrownBy( () -> scope.query()
+				.where( f -> f.queryString().field( field ).matching(
+						String.format( Locale.ROOT, "%s", unParsableValue ) ) )
+				.fetchAllHits()
+		).isInstanceOf( SearchException.class );
+
+	}
+
+	public static List<? extends Arguments> queryStringSyntax() {
+		// String field, String value1, String value2, String noMatch, String unParsableValue
+		TckBackendFeatures backendFeatures = TckConfiguration.get().getBackendFeatures();
+		return List.of(
+				Arguments.of( "integer", "1", "2", "100", "not-an-int" ),
+				Arguments.of( "instant",
+						//"2000-01-01T01\\:01\\:01Z", "2000-02-02T02\\:02\\:02Z", "2222-02-02T02\\:02\\:02Z",
+						backendFeatures.formatForQueryStringPredicate( InstantFieldTypeDescriptor.INSTANCE,
+								Instant.parse( "2000-01-01T01:01:01Z" ) ).replace( ":", "\\:" ),
+						backendFeatures.formatForQueryStringPredicate( InstantFieldTypeDescriptor.INSTANCE,
+								Instant.parse( "2000-02-02T02:02:02Z" ) ).replace( ":", "\\:" ),
+						backendFeatures.formatForQueryStringPredicate( InstantFieldTypeDescriptor.INSTANCE,
+								Instant.parse( "2222-02-02T02:02:02Z" ) ).replace( ":", "\\:" ),
+						"not-an-instant" ),
+				Arguments.of( "localDate", "2000-01-01", "2000-02-02", "2222-02-02", "not-an-localDate" )
+		);
+	}
+
```

Comment:
Could you try to integrate this with `FieldTypeDescriptor`, so that we're sure we're testing all available types, including those that will get added in the future?